tests/providers/test_IBMQ.py

```diff
@@ -0,0 +1,54 @@
+from qiskit import IBMQ
+from qiskit.providers import QiskitBackendNotFoundError
+
+from shor.gates import CNOT, H
+from shor.layers import Qbits
+from shor.operations import Measure
+from shor.providers.qiskit.IBMQ import IBMQProvider
+from shor.quantum import QC
+
+BACKENDS_TO_TEST = [
```

Comment:
I'll note that, as this is a hard-coded list, we should be sure to keep a lookout for any new IBM devices in the future.

Reply:
Totally, we could also put snapshot testing on this instead, so it automatically picks up changes to this list. This would be a nice alerting mechanism for us, so we know when new quantum services have been launched